google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableResult.java

```diff
@@ -28,108 +27,96 @@
 import java.util.Objects;
 import javax.annotation.Nullable;
 
-public class TableResult implements Page<FieldValueList>, Serializable {
+@InternalApi
+@AutoValue
+public abstract class TableResult implements Page<FieldValueList>, Serializable {
 
-  private static final long serialVersionUID = -4831062717210349819L;
+  private static final long serialVersionUID = 1L;
 
-  @Nullable private final Schema schema;
-  private final long totalRows;
-  private final Page<FieldValueList> pageNoSchema;
-  @Nullable private JobId jobId = null;
+  @AutoValue.Builder
+  public abstract static class Builder {
+    /** Sets the free form human-readable reason metadata caching was unused for the job. */
```

Comment:
did your metadata changes in 3133 bleed into this refactor?

Reply:
Good catch. Removed the unnecessary comment.